Assets/MixedRealityToolkit.Services/InputSystem/MixedRealityInputSystem.cs

```diff
@@ -681,30 +699,19 @@ public void RaisePreFocusChanged(IMixedRealityPointer pointer, GameObject oldFoc
         {
             focusEventData.Initialize(pointer, oldFocusedObject, newFocusedObject);
 
-            // Raise Focus Events on the old and new focused objects.
             if (oldFocusedObject != null)
             {
-                ExecuteEvents.ExecuteHierarchy(oldFocusedObject, focusEventData, OnPreFocusChangedHandler);
+                HandleEvent(focusEventData, oldFocusedObject, OnPreFocusChangedHandler);
             }
-
             if (newFocusedObject != null)
             {
-                ExecuteEvents.ExecuteHierarchy(newFocusedObject, focusEventData, OnPreFocusChangedHandler);
+                HandleEvent(focusEventData, newFocusedObject, OnPreFocusChangedHandler);
             }
 
             // Raise Focus Events on the pointers cursor if it has one.
-            if (pointer.BaseCursor != null)
+            if (pointer?.BaseCursor?.GameObjectReference != null)
```

Comment:
This changes the current behavior with consequences I find difficult to predict.

You were saying this change is motivated by InteractionRecevier not receiving focus events for its interactables. Were those interactables under the receiver in the hierarchy? I'm not sure it is meant to receive them otherwise. I they were under the interactable or you think the behavior should be different, could you open an issue for it? 

In general we should try to narrow down the scope of PRs as much as possible and have an issue backing them, to make it easier for reviewers to understand and approve them quickly.

Reply:
InteractionRecevier, as a derivate of InputSystemGlobalListener, registers as a modal listener and therefore receives everything regardless where in the hierarchy it is. All events could be listened to that way, except for the 4 Focus events.
You are correct, I am going to split this up into their own concerns tomorrow. This has gotten consfusing enough already.

Reply:
My point about InteractionReceiver was that if it is meant to have its interactables below it in the hierarchy it would receive those focus events when they bubble up. There is nothing in code enforcing interactables to be below the InteractionReceiver though. @radicalad , do you know if InteractionReceiver is meant to have its interactables as siblings/descendants?

Reply:
InteractionReceiver and the Interactables are entirely different. I agree InteractionReceiver should handle focus, and it did in the [HoloToolkit version](https://github.com/Microsoft/MixedRealityToolkit-Unity/blob/htk_release/Assets/HoloToolkit/UX/Scripts/Receivers/InteractionReceiver.cs).

InteractionReceiver is a easy way to run shared logic on a group of  (I'm not saying _interactable_ since its overloaded) objects in the scene. Simply inherit InteractionReceiver, and override whichever events you desire. Objects in InteractionReceiver's list will bubble up.

Reply:
Thanks for the info @radicalad . So, just to be clear, does InteractionReceiver require the objects it is keeping track of to be descendants of its game object?